`config`: make all `backlog_controller` properties `needs_restart::no`

WillemKauf opened the pull request:
The two use cases of the `backlog_controller` in the code presently are the `compaction_controller` and the `upload_controller`. In both cases, there are cluster properties associated with the PID values used in the `backlog_controller_config`, which presently require a restart to take effect.

There is no strong reason for this to be the case. To enable making these values changeable at run-time, alter the `backlog_controller_config` and use of it in the `backlog_controller` to instead use `config::binding`s around the raw value types.

While this does throw away some of the bare value type simplicity of the `backlog_controller` design, the benefits and actual use cases of this class in the codebase are a good motivation for making this change.

The properties for the `upload_controller` and `compaction_controller` can now be marked as not needing a restart to take effect, since all of their uses are now bindings.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none

WillemKauf committed:
`config`: make PID controller properties `needs_restart::no`

The properties for the `upload_controller` and `compaction_controller`
can now be marked as not needing a restart to take effect, since
all of their uses are now bindings.

WillemKauf committed:
`storage`: remove unused `backlog_controller::update_setpoint()` function

WillemKauf committed:
`storage`: make `backlog_controller_config` properties `config::binding`s

The two use cases of the `backlog_controller` in the code presently
are the `compaction_controller` and the `upload_controller`. In both cases,
there are cluster properties associated with the PID values used in the
`backlog_controller`, which presently require a restart to take effect.

There is no strong reason for this to be the case. To enable making
these values changeable at run-time, alter the `backlog_controller_config`
and use of it in the `backlog_controller` to instead use `config::binding`s
around the raw value types.

While this does throw away some of the bare value type simplicity of the
`backlog_controller` design, the benefits and actual use cases of this
class in the codebase are a good motivation for making this change.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/70126-->
<!--7fd2869c-4270-4f2d-9966-de78c26c8b2c-->
<!--title-->
## Retry command for [Build#70126](https://buildkite.com/redpanda/redpanda/builds/70126)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsMultiClusterTest.test_basic
tests/rptest/tests/datalake/iceberg_rest_catalog_smoke_test.py::IcebergRESTCatalogSmokeTest.test_basic@{"cloud_storage_type":1,"filesystem_catalog_mode":false}
tests/rptest/tests/read_replica_e2e_test.py::TestReadReplicaService.test_writes_forbidden@{"cloud_storage_type":2,"partition_count":10}
tests/rptest/tests/adjacent_segment_merging_test.py::AdjacentSegmentMergingTest.test_reupload_of_local_segments@{"acks":1,"cloud_storage_type":1}
tests/rptest/tests/e2e_shadow_indexing_test.py::ShadowIndexingInfiniteRetentionTest.test_segments_not_deleted@{"cloud_storage_type":1}
tests/rptest/tests/timequery_test.py::TestReadReplicaTimeQuery.test_timequery
tests/rptest/tests/alter_topic_configuration_test.py::ShadowIndexingGlobalConfig.test_topic_manifest_reupload
tests/rptest/tests/retention_policy_test.py::ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden@{"cloud_storage_type":2,"local_retention_ms":-1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_isolate@{"cloud_storage_type":1}
tests/rptest/tests/retention_policy_test.py::ShadowIndexingLocalRetentionTest.test_shadow_indexing_default_local_retention@{"cloud_storage_type":2,"cluster_remote_write":true,"topic_remote_write":null}
tests/rptest/tests/datalake/cluster_restore_test.py::DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq@{"catalog_type":"rest_jdbc","cloud_storage_type":1}
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_spec_evolution@{"catalog_type":"nessie","cloud_storage_type":1}
tests/rptest/tests/archive_retention_test.py::CloudArchiveRetentionTest.test_delete@{"cloud_storage_type":2,"retention_type":"retention.ms"}
tests/rptest/tests/datalake/simple_connect_test.py::RedpandaConnectIcebergTest.test_translating_avro_serialized_records@{"cloud_storage_type":1}
tests/rptest/tests/datalake/datalake_usage_test.py::IcebergUsageTest.test_iceberg_usage@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark"}
tests/rptest/tests/cluster_config_test.py::ClusterConfigAzureSharedKey.test_live_shared_key_change@{"cloud_storage_type":2}
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_chunks
tests/rptest/tests/cloud_topics/e2e_test.py::EndToEndCloudTopicsTxTest.test_write
tests/rptest/tests/datalake/nessie_catalog_smoke_test.py::NessieCatalogSmokeTest.test_nessie_with_trino@{"cloud_storage_type":1}
tests/rptest/tests/consumer_group_recovery_test.py::ConsumerOffsetsRecoveryTest.test_consumer_offsets_partition_recovery@{"force_offset_upload_failures":true}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"include_groups":false,"params":{"cancellation":{"dir":"in","stage":"preparing"},"use_alias":true},"transfer_leadership":false}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"include_groups":true,"params":{"cancellation":{"dir":"in","stage":"executing"},"use_alias":true},"transfer_leadership":false}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"include_groups":false,"params":{"cancellation":null,"use_alias":true},"transfer_leadership":true}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"include_groups":false,"params":{"cancellation":{"dir":"out","stage":"executing"},"use_alias":false},"transfer_leadership":true}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"include_groups":true,"params":{"cancellation":{"dir":"in","stage":"prepared"},"use_alias":true},"transfer_leadership":true}
tests/rptest/tests/datalake/blocked_catalog_test.py::DatalakeBlockedCatalogTest.test_trim_local_after_first_translation@{"cloud_storage_type":1}
tests/rptest/tests/datalake/datalake_dlq_test.py::DatalakeDLQTest.test_dlq_table_for_invalid_records@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark"}
tests/rptest/tests/datalake/datalake_dlq_test.py::DatalakeDLQTest.test_dlq_table_for_mixed_records@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"query_engine":"trino"}
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_e2e_basic@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"query_engine":"trino"}
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_remove_expired_snapshots@{"catalog_type":"rest_jdbc","cloud_storage_type":1}
tests/rptest/tests/datalake/translation_interruption_test.py::DatalakeTranslationInterruptionsTest.test_oom_interruptions@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark"}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndShadowIndexingTest.test_reset_spillover@{"cloud_storage_type":2}
tests/rptest/tests/tiered_storage_model_test.py::TieredStorageTest.test_tiered_storage@{"cloud_storage_type_and_url_style":[2,"virtual_host"],"test_case":{"name":"(TS_Read == True, AdjacentSegmentMergerReupload == True)"}}
tests/rptest/tests/tiered_storage_model_test.py::TieredStorageTest.test_tiered_storage@{"cloud_storage_type_and_url_style":[1,"path"],"test_case":{"name":"(TS_Read == True, AdjacentSegmentMergerReupload == True)"}}
tests/rptest/tests/tiered_storage_model_test.py::TieredStorageTest.test_tiered_storage@{"cloud_storage_type_and_url_style":[1,"virtual_host"],"test_case":{"name":"(TS_Read == True, AdjacentSegmentMergerReupload == True)"}}
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_prevent_recovery@{"cloud_storage_type":1}
tests/rptest/tests/timequery_test.py::TimeQueryTest.test_timequery_with_local_gc
tests/rptest/tests/cluster_metadata_upload_test.py::ClusterMetadataUploadTest.test_uploads_after_wipe@{"cloud_storage_type":1}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_listing_inexistent_migration
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_avro_schema@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark"}
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_iceberg_partition_key_file_location@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"custom_partition_spec":"(number)"}
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_json_schema_unicode@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"query_engine":"trino"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_select_fails@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"avro","query_engine":"spark"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_no_collision@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"avro","query_engine":"spark"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_select_fails@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_no_collision@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_illegal_schema_evolution@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"spark","test_case":"drop column that appears in partition spec"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_illegal_schema_evolution@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"spark","test_case":"illegal promotion int->string"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_illegal_schema_evolution@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"trino","test_case":"drop column that appears in partition spec"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_illegal_schema_evolution@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"trino","test_case":"illegal promotion int->string"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"spark","test_case":"add_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"avro","query_engine":"spark","test_case":"promote_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"avro","query_engine":"spark","test_case":"reorder_columns"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino","test_case":"add_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino","test_case":"drop_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino","test_case":"promote_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_legal_schema_evolution@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino","test_case":"reorder_columns"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"spark","test_case":"add_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"spark","test_case":"drop_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"proto3","query_engine":"spark","test_case":"promote_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino","test_case":"add_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino","test_case":"drop_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"avro","query_engine":"trino","test_case":"promote_column"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer@{"catalog_type":"nessie","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino","test_case":"reorder_columns"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_reorder_columns@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"spark"}
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_reorder_columns@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"proto2","query_engine":"trino"}
tests/rptest/tests/shadow_indexing_admin_api_test.py::SIAdminApiTest.test_bucket_validation@{"cloud_storage_type":2}
tests/rptest/tests/shadow_indexing_firewall_test.py::ShadowIndexingFirewallTest.test_consume_from_blocked_s3@{"cloud_storage_type":2}
tests/rptest/tests/self_test_test.py::SelfTestTest.test_self_test_unknown_test_type
tests/rptest/tests/topic_creation_test.py::RapidTopicRecreateTest.test_topic_rapid_recreation
```
<!--/body-->

WillemKauf commented on src/v/redpanda/application.cc:
```diff
@@ -1215,15 +1216,18 @@ static storage::backlog_controller_config
-        (space_info.capacity / 100) * backlog_capacity_percents
-        / ss::smp::count);
+
+    auto backlog_size_function = [capacity = space_info.capacity] {
```
I wonder if this (like the `upload_controller_config`) should use available disk space as a setpoint/normalization factor instead of capacity 🤔

WillemKauf commented on src/v/redpanda/application.cc:
```diff
@@ -1269,22 +1273,23 @@ make_upload_controller_config(ss::scheduling_group sg) {
-      = ss::fs_avail(config::node().data_directory().path.string()).get();
-    int64_t setpoint = 0;
-    int64_t normalization = static_cast<int64_t>(available)
+    auto space_info = std::filesystem::space(
```
needs to be a synchronous call here.

oleiman replied:
q: why is this preferable to `ss::fs_avail`?

WillemKauf replied:
we were using `ss:fs_avail().get()` before, which was fine in the `async` context within `application.cc`, but now that this is a sharded parameter, `.get()` will fail outside of that context.

`std::filesystem::space` is a synchronous call for the same information.

dotnwat replied:
> needs to be a synchronous call here.

i don't really understand what's happening here.

> we were using ss:fs_avail().get() before, which was fine in the async context within application.cc, but now that this is a sharded parameter, .get() will fail outside of that context.

but couldn't we call this outside and pass in the result?

WillemKauf replied:
>i don't really understand what's happening here.

A swap from `ss::fs_avail().get()` to `std::filesystem::space()` to probe the underlying disk for its capacity, that's all.

> but couldn't we call this outside and pass in the result?

Sure, there are probably other ways, like capturing the result of `ss::fs_avail()` outside the sharded parameter lambda and passing it to the `backlog_controller_config()` function, but I think this is fine too.

dotnwat replied:
making a synchronous I/O system call isn't ideal (reactor stall risk). i agree it's not a big deal (e.g. just on bootup?), but avoiding it also pretty easy?

WillemKauf replied:
Reverted this to an outside `ss::fs_avail().get()` call that is passed to the config generating functions.

oleiman commented on src/v/redpanda/application.cc:
```diff
@@ -1269,22 +1273,23 @@ make_upload_controller_config(ss::scheduling_group sg) {
-    int64_t normalization = static_cast<int64_t>(available)
+    auto space_info = std::filesystem::space(
+      config::node().data_directory().path);
+    auto setpoint_function = []() { return 0; };
```
this has kind of a funny smell. could probably make a case for leaving the setpoint as a raw value and requiring restart for the compaction backlog config. that is, I think the ability to tune a controller live is still useful independent of the setpoint.

is changing the compaction compaction backlog target something that people are clambering for currently?

WillemKauf replied:
> this has kind of a funny smell.

I agree, it's a bit odd, but its mostly to support the logic for the backlog within `compaction_controller` (which preferentially uses a cluster config that is an optional, but falls back to a default based on available disk capacity otherwise). I'm not sure if there's a better/cleaner solution here (while keeping setpoint tunable).

> I think the ability to tune a controller live is still useful independent of the setpoint... is changing the compaction compaction backlog target something that people are clambering for currently?

I wouldn't say anyone is clambering for any of these changes, but I am also going to argue that _all_ cluster properties should be `needs_restart::no` if it is possible, given how intrusive broker restarts have proven to be from a customer facing point of view. I think there should be very strong reasons as to why a cluster config is `needs_restart::yes`, because you never know when it will be useful in the future to be able to flip a setting on the fly.

oleiman reviewed (commented):
looks legit. couple questions mostly for my own edification

dotnwat commented on src/v/storage/backlog_controller.cc:
```diff
@@ -82,29 +86,34 @@ ss::future<> backlog_controller::stop() {
 ss::future<> backlog_controller::update() {
-    _current_backlog = co_await _sampler->sample_backlog() / _norm;
-    auto current_err = _setpoint - _current_backlog;
+    auto norm = _cfg.normalization_factor;
+    auto setpoint = _cfg.setpoint_cb() / norm;
+    _current_backlog = co_await _sampler->sample_backlog() / norm;
+
+    auto current_err = setpoint - _current_backlog;
```
there is a lot going on in this commit. switching to cfg bindings, async/sync filesystem space query, some refactoring (it looks like) maybe it can be split up?

WillemKauf replied:
Yeah, I agree it seems like a lot for one commit, but due to the underlying type swapping to bindings I think most every change here is necessary in one commit to ensure every commit builds.

dotnwat replied:
ok. ill give it another look then tomorrow

WillemKauf replied:
Moved async/sync filesystem space query change out of this commit.

WillemKauf replied:
@dotnwat lmk if this looks good to you.

dotnwat reviewed (commented):
nice change to make it runtime configurable. maybe the 2nd commit can be split up?

WillemKauf committed:
`redpanda`: pass disk availability to `controller_config()` functions

This also has the functional change of using availability instead of capacity
as the factor by which the normalization factor and setpoint is calculated
for the `compaction_controller_config`.

This makes more logical sense, since we should only be concerned with disk
space available to `redpanda` as an application instead of the disk's total
capacity when trying to make informed decisions on space management internally.

WillemKauf commented:
Force push to:

* Add commit which uses `ss::fs_avail()` as argument to `controller_config()` functions, and replace use of `std::filesystem::space()`.

oleiman reviewed (dismissed):
lgtm

dotnwat commented on src/v/redpanda/application.cc:
```diff
@@ -1976,6 +1971,8 @@ void application::wire_up_redpanda_services(
       std::ref(feature_table))
       .get();
 
+    auto fs_avail
+      = ss::fs_avail(config::node().data_directory().path.string()).get();
```
it looks like this commit is swapping out capacity for free space, which if true would be a pretty big difference but there isn't a mention of that. am i reading it correctly?

WillemKauf replied:
You are, [I mentioned it in a comment above](https://github.com/redpanda-data/redpanda/pull/27106#discussion_r2249445267).

I think using availability here makes more sense than capacity, no?

dotnwat replied:
please explain in the commit message

dotnwat replied:
someone should be able to use git-blame and get an idea of what's going on in the change.

WillemKauf replied:
Added color to this commit message: https://github.com/redpanda-data/redpanda/pull/27106/commits/6bccb63f73f8f7eb2e18d1fd92a808ed737afa1e.

>This also has the functional change of using availability instead of capacity
as the factor by which the normalization factor and setpoint is calculated
for the `compaction_controller_config`.
> This makes more logical sense, since we should only be concerned with disk
space available to `redpanda` as an application instead of the disk's total
capacity when trying to make informed decisions on space management internally.

dotnwat replied:
thanks!

dotnwat commented on src/v/redpanda/application.cc:
```diff
@@ -1267,20 +1270,21 @@ make_upload_controller_config(ss::scheduling_group sg, uint64_t fs_avail) {
-      0,
-      config::shard_local_cfg().cloud_storage_upload_ctrl_d_coeff(),
+      config::shard_local_cfg().cloud_storage_upload_ctrl_p_coeff.bind(),
+      config::mock_binding(0.0),
```
why was the integral component here fixed at 0 (and not a real config binding?)

WillemKauf replied:
https://github.com/redpanda-data/redpanda/blob/6aedc4e40a7e48f64ec8111559c3901daa3855ec/src/v/redpanda/application.cc#L1263-L1271

dotnwat reviewed (approved):
very nice. sorry for the review delay!

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--1c64e684-1844-4e7d-98d5-23a9a817e7c8-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70835">build#70835</a></summary>


| test_class               | test_method                 | test_arguments                                                                                                                           | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:-------------------------|:----------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| MasterTestSuite          | test_replica_pair_frequency |                                                                                                                                          | unit        | https://buildkite.com/redpanda/redpanda/builds/70835#0198aa59-f9bd-4638-8bb6-0988ee61c3ac | FAIL          | 0/1      |                                                                                                                                                           |
| RandomNodeOperationsTest | test_node_operations        | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": false, "mixed_versions": false, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/70835#0198aaca-10ca-416d-afce-e6fae713dcca | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->
